lib/components/sprite_animation_component.dart

```diff
@@ -15,7 +15,7 @@ class SpriteAnimationComponent extends PositionComponent {
     Vector2 size,
     this.animation, {
     this.destroyOnFinish = false,
-  }) {
+  }) : assert(animation != null) {
```

Comment:
Can't you set `this.animation` as `@required` and it is automatically not nullable?

Reply:
No :(, `@required` are only available to named parameters

Reply:
@required doesn't make it non-nullable, it makes it required (named parameters can be omitted). You can still explicitly pass in null. for real null safety we need to unable the new dart versio.